esp-hal-common/src/uart.rs

```diff
@@ -396,6 +408,39 @@ where
             Err(nb::Error::WouldBlock)
         }
     }
+
+    /// Read all available bytes from the RX FIFO into the provided buffer and
+    /// returns the number of read bytes. Never blocks
+    pub fn drain_fifo(&mut self, buf: &mut [u8]) -> usize {
+        #[allow(unused_variables)]
+        let offset = 0;
+
+        // on ESP32-S2 we need to use PeriBus2 to read the FIFO
+        #[cfg(esp32s2)]
+        let offset = 0x20c00000;
+
+        let mut count = 0;
+        while T::get_rx_fifo_count() > 0 && count < buf.len() {
+            let value = unsafe {
+                let fifo = (T::register_block().fifo.as_ptr() as *mut u8).offset(offset)
+                    as *mut crate::peripherals::generic::Reg<FIFO_SPEC>;
+                (*fifo).read().rxfifo_rd_byte().bits()
+            };
+            buf[count] = value;
+            count += 1;
+        }
+        count
+    }
+
+    // Returns whether the fifo rx timeout feature is enabled
+    fn rx_timeout_enabled(&self) -> bool {
```

Comment:
Instead of a checking the registers, why don't we store the rx config, and optionally return that if its been configured? That way we can also see what the timeout is configured to.

Reply:
Done